dlc-manager/src/manager.rs

```diff
@@ -705,14 +702,8 @@ where
                     "Previous tx output not found at index {}",
                     vout
                 )))?;
-            dlc::util::sign_p2wpkh_input(
-                &self.secp,
-                &sk,
-                &mut fund_tx,
-                input_index,
-                bitcoin::SigHashType::All,
-                tx_out.value,
-            );
+            self.wallet
+                .sign_for_witness(&self.secp, &mut fund_tx, input_index, tx_out)?;
```

Comment:
You're passing `fund_tx` as `&mut` but `sign_for_witness` doesn't take a mutable reference, and also it doesn't actually sign the transaction but only generates the witness AFAIU (in bitcoin-rpc-provider). So one solution is to make `fund_tx` mutable in the trait and update bitcoin-rpc-provider to actually sign.

Reply:
Also I don't know how the code will look like once you update, but it might be easier to just sign within the wallet and then extract the witness from the signed tx after?

Reply:
You're still not signing the transaction AFAICT. Now you're only returning the witness, but the original transaction is not modified (unfortunately the CI is not perfect and just get stuck when issues happen on the receiving thread, you can use e.g. `cargo test --all-features -- --ignored two_of_five_oracle_numerical_test --nocapture` to see the issue locally. For now I get:
```
thread '<unnamed>' panicked at 'Unexpected error Wallet error Bitcoin Rpc Error JSON-RPC error: RPC error response: RpcError { code: -22, message: "Previous output scriptPubKey mismatch:\n0 f5260f9b461ae6faf3b7effc71c2323430aa072c\nvs:\n0 a78929ab9d88c7c632b32d9197fac277dcb0d742", data: None }', dlc-manager/tests/manager_execution_tests.rs:757:24
```
which looks like bitcoin core wallet is not super happy. Though I must say I really like using `sign_raw_transaction_with_wallet`. I think you have a couple of ways to do it:
* have the `sign` method take ownership of the fund tx and return the signed tx, then extract the witness when needed
* have the `sign` method take a mutable reference, then add the witness within it. Then you can either return an empty result or the witness (I think empty result might be cleaner).
* have the `sign` method take an immutable reference and only return the witness

Whichever would work I think, I think I'd prefer one of the first two but I don't have a strong opinion.

Reply:
I am very new to rust, but 1 & 2 seem much more of a C++ pattern vs 3 is much better for functional programming

Reply:
I can't find where the bug is coming from, is it possible the tests are using dummy data that bitcoin core is validating against? It seems to only be happening in the on_sign function

Reply:
Went for option 2